src/sonic-yang-models/yang-models/sonic-pbh.yang

```diff
@@ -0,0 +1,218 @@
+						pattern "INNER_IP_PROTOCOL|INNER_L4_DST_PORT|INNER_L4_SRC_PORT|INNER_DST_IPV4|INNER_SRC_IPV4|INNER_DST_IPV6|INNER_SRC_IPV6";
+					}
+				}
+
+				leaf ipv4_mask {
+					description "Configures IPv4 address mask for this hash";
+					mandatory true;
+					type inet:ipv4-address;
+					when "boolean(../hash_field[.='INNER_DST_IPV4' or .='INNER_SRC_IPV4'])";
+				}
+
+				leaf ipv6_mask {
+					description "Configures IPv6 address mask for this hash";
+					mandatory true;
+					type inet:ipv6-address;
+					when "boolean(../hash_field[.='INNER_DST_IPV6' or .='INNER_SRC_IPV6'])";
+				}
+
+				leaf sequence_id {
+					description "Configures in which order the fields are hashed and defines which fields should be associative";
+					mandatory true;
+					type uint32;
+				}
+
+			}
+			/* end of PBH_HASH_LIST */
+		}
+		/* end of container PBH_HASH */
+
+		container PBH_RULE {
+
+			description "PBH_RULE part of config_db.json";
+
+			list PBH_RULE_LIST {
+
+				key "table_name rule_name";
+
+				leaf table_name {
+					description "The name of table which holds this rule";
+					type leafref {
+						path "/pbh:sonic-pbh/pbh:PBH_TABLE/pbh:PBH_TABLE_LIST/pbh:table_name";
+					}
+				}
+
+				leaf rule_name {
+					description "The name of this rule";
+					type string;
+				}
+
+				leaf priority {
+					description "Configures priority for this rule";
+					mandatory true;
+					type uint32;
+				}
+
+				leaf gre_key {
+					description "Configures packet match for this rule: GRE key";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf ip_protocol {
+					description "Configures packet match for this rule: IP protocol";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf ipv6_next_header {
+					description "Configures packet match for this rule: IPv6 Next header";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf l4_dst_port {
+					description "Configures packet match for this rule: L4 destination port";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf inner_ether_type {
+					description "Configures packet match for this rule: inner EtherType";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf-list hash_list {
+					description "The list of hash fields to apply with this rule";
+					min-elements 1;
+					type leafref {
+						path "/pbh:sonic-pbh/pbh:PBH_HASH/pbh:PBH_HASH_LIST/pbh:hash_name";
+					}
+				}
+
+				leaf packet_action {
+					description "Configures packet action for this rule";
+					type string {
+						pattern "SET_ECMP_HASH|SET_LAG_HASH";
+					}
+					default "SET_ECMP_HASH";
+				}
+
+				leaf flow_counter {
+					description "Enables/Disables packet/byte counter for this rule";
+					type string {
+						pattern "DISABLED|ENABLED";
```

Comment:
The type definition for this (DISABLED/ENABLED) can go to "sonic-types.yang" file

Reply:
@venkatmahalingam we can do it once there will be another schema which requires it